hoomd/md/pytest/test_methods.py

```diff
@@ -3,393 +3,303 @@
+                                                    hoomd.md.methods.NPT)])
+
+    nvt_setup_params = {'kT': hoomd.variant.Constant(2.0), 'tau': 2.0 }
+    nvt_extra_params = { }
+    nvt_accepted_params = {'kT': hoomd.variant.Ramp(1, 2, 1000000, 2000000), 'tau': 10.0,
+            'translational_thermostat_dof': (0.125, 0.5), 'rotational_thermostat_dof': (0.5, 0.25)}
+    nvt_rejected_params = { }
+
+    method_base_params_list.extend([paramtuple(nvt_setup_params,
+                                                    nvt_extra_params,
+                                                    nvt_accepted_params,
+                                                    nvt_rejected_params,
+                                                    hoomd.md.methods.NVT)])
+
+    nve_setup_params = { }
+    nve_extra_params = { }
+    nve_accepted_params = { }
+    nve_rejected_params = { }
+
+    method_base_params_list.extend([paramtuple(nve_setup_params,
+                                                    nve_extra_params,
+                                                    nve_accepted_params,
+                                                    nve_rejected_params,
+                                                    hoomd.md.methods.NVE)])
+
+    return method_base_params_list
+
+
+@pytest.fixture(scope="function", params=_method_base_params(), ids=(lambda x: x[4].__name__))
+def method_base_params(request):
+    return deepcopy(request.param)
+
+def test_attributes(method_base_params):
 
-    assert npt.translational_thermostat_dof == (0.0, 0.0)
-    npt.translational_thermostat_dof = (0.125, 0.5)
-    assert npt.translational_thermostat_dof == (0.125, 0.5)
-
-    assert npt.rotational_thermostat_dof == (0.0, 0.0)
-    npt.rotational_thermostat_dof = (0.5, 0.25)
-    assert npt.rotational_thermostat_dof == (0.5, 0.25)
-
-    assert npt.barostat_dof == (0.0, 0.0, 0.0, 0.0, 0.0, 0.0)
-    npt.barostat_dof = (1.0, 2.0, 4.0, 6.0, 8.0, 10.0)
-    assert npt.barostat_dof == (1.0, 2.0, 4.0, 6.0, 8.0, 10.0)
-
-
-def test_nph_attributes():
-    """Test attributes of the NPH integrator before attaching."""
     all_ = hoomd.filter.All()
-    constant_s = [
-        hoomd.variant.Constant(1.0),
-        hoomd.variant.Constant(2.0),
-        hoomd.variant.Constant(3.0),
-        hoomd.variant.Constant(0.125),
-        hoomd.variant.Constant(.25),
-        hoomd.variant.Constant(.5)
-    ]
-    nph = hoomd.md.methods.NPH(filter=all_,
-                               S=constant_s,
-                               tauS=2.0,
-                               couple='xyz')
-
-    assert nph.filter == all_
-    assert len(nph.S) == 6
-    for i in range(6):
-        assert nph.S[i] is constant_s[i]
-    assert nph.tauS == 2.0
-    assert nph.box_dof == (True, True, True, False, False, False)
-    assert nph.couple == 'xyz'
-    assert not nph.rescale_all
-    assert nph.gamma == 0.0
+    integrator = method_base_params.method(**method_base_params.setup_params,filter=all_)
```

Comment:
Use `method` for the variable name here and elsewhere to not confuse the instance with a HOOMD-blue style integrator.